src/users/users.controller.ts

```diff
@@ -44,6 +47,17 @@ export class UsersController {
     return this.usersService.findOne(id);
   }
 
+  @UseInterceptors(ClassSerializerInterceptor)
+  @Patch('/me/:id')
+  @UseGuards(AuthGuard, RolesGuard)
+  @Roles(Role.USER)
```

Comment:
Here the Roles permitted should be for user and admin, since they also should be able to change their password.
`@Roles(Role.USER, Role.ADMIN)`

Reply:
Updated allowed roles